unit_test/extensions/onnx/CMakeLists.txt

```diff
@@ -0,0 +1,47 @@
+
+# If compiling on windows add the stdafx file
+add_executable(rltest-onnx
+  main.cc
+  tensor_notation_test.cc
+  mnist_inference_test.cc
+  mock_helpers.cc
+)
+
+set(ONNX_EXTENSION_TEST_RESOURCE_FILES
+  ${CMAKE_CURRENT_SOURCE_DIR}/mnist_data/mnist_test_data.txt
+  ${CMAKE_CURRENT_SOURCE_DIR}/mnist_data/mnist_model.onnx
+)
+
+add_custom_command(
+  TARGET rltest-onnx
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different
+          ${ONNX_EXTENSION_TEST_RESOURCE_FILES}
+          ${CMAKE_CURRENT_BINARY_DIR}/mnist_data
+)
+
+# Devirtualization is an optimization that changes the vtable if the compiler decides a function
+# doesn't need to be virtual. This is incompatible with the mocking framework used in testing as it
+# makes the vtable structure unpredictable
+if(MSVC)
+  # MSVC doesn't optimize aggresively enough for this bug to be an issue right now.
+else()
+  target_compile_options(rltest-onnx PRIVATE -fno-devirtualize)
```

Comment:
I wonder if gmock would be affected by this?

Reply:
As of this PR the flag shouldn't be needed https://github.com/VowpalWabbit/reinforcement_learning/pull/114